src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -940,43 +950,66 @@ - (void)_handleSubscriptionError:(NSError *)error
 
 // This method is used for signaling whether to use the subscription pool. This functions as
 // a heuristic for whether to throttle subscriptions to the device via a pool of subscriptions.
-// If products appear that have both Thread and Wifi enabled but is primarily on wifi, this
+// If products appear that have both Thread and Wifi enabled but are primarily on wifi, this
 // method will need to be updated to reflect that.
 - (BOOL)_deviceUsesThread
 {
     os_unfair_lock_assert_owner(&self->_lock);
 
-    // Device is thread-enabled if there is a Thread Network Diagnostics cluster on endpoint 0
-    for (MTRClusterPath * path in [self _knownClusters]) {
-        if (path.endpoint.unsignedShortValue != 0) {
-            continue;
+#ifdef DEBUG
+    id testDelegate = _weakDelegate.strongObject;
+    if (testDelegate) {
+        if ([testDelegate respondsToSelector:@selector(unitTestPretendThreadEnabled:)]) {
+            if ([testDelegate unitTestPretendThreadEnabled:self]) {
+                return YES;
+            }
         }
+    }
+#endif
 
-        if (path.cluster.unsignedLongValue == MTRClusterIDTypeThreadNetworkDiagnosticsID) {
-            return YES;
-        }
+    MTRClusterPath * networkCommissioningClusterPath = [MTRClusterPath clusterPathWithEndpointID:@(0) clusterID:@(MTRClusterIDTypeNetworkCommissioningID)];
+    MTRDeviceClusterData * networkCommissioningClusterData = [self _clusterDataForPath:networkCommissioningClusterPath];
+    NSNumber * networkCommissioningClusterFeatureMapValueNumber = networkCommissioningClusterData.attributes[@(MTRClusterGlobalAttributeFeatureMapID)][MTRValueKey];
```

Comment:
Will this do the right things when `networkCommissioningClusterData` is nil or `networkCommissioningClusterData.attributes[@(MTRClusterGlobalAttributeFeatureMapID)]` is nil?  That is, when we don't have the information yet?

Seems like at best it will make `networkCommissioningClusterFeatureMapValueNumber` nil which will cause `isKindOfClass` to be false, and then we get the MTR_LOG_ERROR... but that's not really an error condition, right?